src/components/cdPipeline/BuildCD.tsx

```diff
@@ -392,6 +386,66 @@ export default function BuildCD({
         const handleFormatHighlightedText = (opt: Environment, { inputValue }) => {
             return formatHighlightedTextDescription(opt, inputValue, 'name')
         }
+
+        return (
+            <>
+                <div className="form__label dc__required-field">Environment</div>
+                <ReactSelect
+                    menuPosition={isAdvanced ? null : 'fixed'}
+                    closeMenuOnScroll
+                    isDisabled={!!cdPipelineId}
+                    classNamePrefix="cd-pipeline-environment-dropdown"
+                    placeholder="Select Environment"
+                    options={releaseMode === ReleaseMode.LINK ? envList.filter((env) => !env.isVirtualEnvironment) : envList}
+                    value={selectedEnv}
+                    getOptionLabel={(option) => `${option.name}`}
+                    getOptionValue={(option) => `${option.id}`}
+                    isMulti={false}
+                    onChange={(selected: any) => selectEnvironment(selected)}
+                    components={{
+                        IndicatorSeparator: null,
+                        DropdownIndicator,
+                        SingleValue: singleOption,
+                        GroupHeading: groupHeading,
+                    }}
+                    styles={{
+                        ...groupStyle(),
+                        control: (base) => ({ ...base, border: '1px solid #d6dbdf', minHeight: 36, height: 36 }),
+                    }}
+                    formatOptionLabel={handleFormatHighlightedText}
+                />
+                {isEnvUsedState && (
+                    <span className="form__error">
+                        <img src={error} className="form__icon" />
+                        {ENV_ALREADY_EXIST_ERROR}
+                    </span>
+                )}
+                {!formDataErrorObj.envNameError.isValid ? (
+                    <span className="form__error">
+                        <AlertTriangle className="icon-dim-14 mr-5 ml-5 mt-2" />
+                        {formDataErrorObj.envNameError.message}
+                    </span>
+                ) : null}
+                {renderVirtualEnvironmentInfo()}
+            </>
+        )
+    }
+
+    const renderEnvNamespaceAndTriggerType = () => {
+        const envId = formData.environmentId
+        const selectedEnv: Environment = formData.environments.find((env) => env.id == envId)
+        const namespaceEditable = false
+
+        const getNamespaceplaceholder = (): string => {
+            if (isVirtualEnvironment) {
```

Comment:
Can we move this method outside this method it is anti pattern to have methods in methods